pkg/controller/nodemaintenance/nodemaintenance_controller.go

```diff
@@ -4,33 +4,133 @@ package nodemaintenance
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
+	"math"
+	"net"
+	"strconv"
 	"time"
 
 	log "github.com/sirupsen/logrus"
+
+	coordv1beta1 "k8s.io/api/coordination/v1beta1"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/fields"
-	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/types"
-	kubernetes "k8s.io/client-go/kubernetes"
-	"k8s.io/client-go/rest"
-	"k8s.io/client-go/tools/cache"
-	"k8s.io/kubernetes/pkg/kubectl/drain"
-	kubevirtv1alpha1 "kubevirt.io/node-maintenance-operator/pkg/apis/kubevirt/v1alpha1"
+	apitypes "k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller"
-	"sigs.k8s.io/controller-runtime/pkg/event"
 	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
-	"sigs.k8s.io/controller-runtime/pkg/predicate"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/strategicpatch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	drain "k8s.io/kubectl/pkg/drain"
 )
 
-// Add creates a new NodeMaintenance Controller and adds it to the Manager. The Manager will set fields on the Controller
-// and Start it when the Manager is Started.
+var _ reconcile.Reconciler = &ReconcileNodeMaintenance{}
+
+const (
+	NodeMaintenanceSpecAnnotation   = "lifecycle.openshift.io/maintenance"
+	NodeMaintenanceStatusAnnotation = "lifecycle.openshift.io/maintenance-status"
+	LeaseHolderIdentity             = "node-maintenance"
+	LeasePaddingSeconds             = 30
+	WaitForLeasePeriod              = 10 * time.Second
+)
+
```

Comment:
If any of these constants are subjected to user config then i would suggest adding them as a field under the maintenance CRD (can be done in a different PR)